app/authentication/user_id_generator.py

```diff
@@ -15,28 +14,28 @@
 
 class UserIDGenerator(object):
 
-    @staticmethod
-    def generate_id(metadata):
+    def __init__(self, iterations):
+        self.iterations = iterations
+
+    def generate_id(self, metadata, salt):
         collection_exercise_sid, eq_id, form_type, ru_ref = UserIDGenerator._get_token_data(metadata)
```

Comment:
Worth adding a None check for metadata and salt and throwing `ValueError` (plus unit test) as this is critical stuff